### ISSUE TITLE
How to get read counts for all variants relative to reference?

### ISSUE THREAD

**Issue by iaincambeul** [barricklab/breseq#282]
Hello,

This may be a rudimentary question, but I was wondering if there was a way in breseq to output the number of reads supporting each mutation (or combination of mutations) while running in polymorphism mode?  I have a large number of mutations I am looking to track across my samples and even when I remove all filters, as suggested in the tutorials, I only see a subset of the mutations (those with most depth support).  Additionally, I am not sure how to extract a table relating reads to predicted mutations, even for the predicted mutations I do observe.  

This is an example of what I have been running
`
breseq -p --polymorphism-reject-indel-homopolymer-length 0 --polymorphism-reject-surrounding-homopolymer-length 0 --polymorphism-bias-cutoff 0 --polymorphism-frequency-cutoff 0 -o no-filter -r REFERENCE.fasta  READS.fastq.gz -j 6
`

**Comment by jeffreybarrick** [comment]
Here are two things to know to help you get the info that you'd like:

**Part 1:** If there are no reads supporting the variant (mutation) then _breseq_ won't show/tabulate the coverage information by default. (It would have to do it for every position in the genome since it doesn't know what is important.) 

The way to force it to provide this information for all samples is to first create a merged GenomeDiff file from the output of all of your initial runs on the separate samples. You can do this using a command like this:
```
gdtools UNION -e -o merged.gd input1.gd input2.gd input3.gd ...
```
You need the `-e` option or it will remove the RA evidence lines that are needed in the next step.

Now re-run _breseq_ on all of your samples into clean output directories, adding this argument:
`--user-evidence-gd merged.gd`

In the new output, you should now have mutation entries in the HTML for each sample, even for mutations at 0% frequency.

---

**Part 2:** The second part of your question is how to get a count of the reads supporting each mutation in each sample.

For this, the information is provided in the `RA` lines of the output GenomeDiff files.

They have the following key value pairs on the line (and many others):
```
ref_cov=0/1
new_cov=27/42
```
The values here are the number of reads supporting the reference ("ref") or variant ("new") base on the top strand / bottom strand (the two numbers separated by the slash).

One way to access that information so you can use it in other programs is to convert the GD file using this command:
```
gdtools ANNOTATE -p -f TSV -o output.tsv input.gd
```

This will convert a GenomeDiff file to a tab-separated values file with these keys converted into `ref_cov` and `new_cov` columns. Then, you can use the values in these columns (after separating the top/bottom strands) in whatever program you are using.

_Hopefully, this makes sense! Let me know if you run into issues trying to put it into action_

**Comment by iaincambeul** [comment]
Thanks for the very fast response.  That is very helpful and I just have a few followup questions:

What does the frequency in the mutation prediction table represent?  Frequency of a mutation at that site? Or frequency of _that_ mutation at _that_ site?

Also I have tried running

`
gdtools ANNOTATE -p -f TSV -r ref.fasta -o output.tsv ra_mc_evidence.gd
`

And have gotten empty .tsv files on a couple of .gd files that I have verified are non-empty and have predicted mutations in their contents.  I added the -r argument since the script gave me the help menu everytime I failed to include it.  Is there something I'm missing about how to reformat .gd to .tsv?

**Comment by jeffreybarrick** [comment]
With #1, you've hit upon a fundamental limitation of _breseq_'s current assumptions. It assumes there are only two "real" bases at any given position when calculating frequencies: the reference base and the top variant base. There is also a `total_cov` field that will give you the sum of those plus any bases other than those two.

For #2, you're right, you do need the `-r` option. For your problem of empty output... does your input GenomeDiff file have RA lines? Maybe send me an example with your `ref.fasta` at the email address in the _breseq_ header.

**Comment by jeffreybarrick** [comment]
Ah.. I think I see the problem.

The correct flag for preserving evidence is `-e` or `--preserve-evidence`, so try:

```
gdtools ANNOTATE -e -f TSV -r ref.fasta -o output.tsv ra_mc_evidence.gd
```

Also, know that currently `JC` evidence is not output by this command due to a bug. I'm not sure if that also applies to `MC` evidence, but it will be fixed in an update to v0.36.0 that I will put out shortly.

**Comment by iaincambeul** [comment]
Worked perfectly.  Thanks for the repeated help!
